modules/backendscheduler/provider/compaction.go

```diff
@@ -93,33 +92,46 @@ func (p *CompactionProvider) Start(ctx context.Context) <-chan *work.Job {
 		level.Info(p.logger).Log("msg", "compaction provider started")
 
 		var (
-			job *work.Job
-			b   = backoff.New(ctx, p.cfg.Backoff)
+			job               *work.Job
+			b                 = backoff.New(ctx, p.cfg.Backoff)
+			curTenantJobCount int
 		)
 
 		reset := func() {
 			metricTenantReset.WithLabelValues(p.curTenant.Value()).Inc()
 			p.curSelector = nil
 			p.curTenant = nil
-			p.curTenantJobCount = 0
+			curTenantJobCount = 0
 		}
 
 		for {
+			if ctx.Err() != nil {
+				level.Info(p.logger).Log("msg", "compaction provider stopping")
+				return
+			}
+
 			if p.curSelector == nil {
 				if !p.prepareNextTenant(ctx) {
+					level.Info(p.logger).Log("msg", "received empty tenant", "waiting", b.NextDelay())
+					metricTenantBackoff.Inc()
 					b.Wait()
 				}
 				continue
 			}
 
-			if p.curTenantJobCount >= p.cfg.MaxJobsPerTenant {
+			if curTenantJobCount >= p.cfg.MaxJobsPerTenant {
+				level.Info(p.logger).Log("msg", "max jobs per tenant reached, skipping to next tenant")
 				reset()
 				continue
 			}
 
 			job = p.createJob(ctx)
 			if job == nil {
+				level.Info(p.logger).Log("msg", "tenant exhausted, skipping to next tenant after delay", "waiting", b.NextDelay())
 				// we don't have a job, reset the curTenant and try again
+				metricTenantNilJob.Inc()
```

Comment:
this is basically indicating the workers to backoff because we have no jobs, let's rename this to something like BackoffJob because we are telling workers to backoff.

Reply:
I pushed a commit to call it EmptyJob.  We're indicating that work was not found to hand out.  The backoff is a side-effect in this case, but its the job creation engine we want to measure here.